backend/src/main/java/com/example/worksync/service/CommentService.java

```diff
@@ -61,4 +63,21 @@ private CommentDTO convertToDTO(Comment comment) {
                 comment.getCreatedAt()
         );
     }
+
+    public void deleteComment(Long commentId, User user) {
+        Optional<Comment> commentOpt = commentRepository.findById(commentId);
+        
+        if (commentOpt.isEmpty()) {
+            throw new ResourceNotFoundException("Comment", commentId);
+        }
+
+        Comment comment = commentOpt.get();
+        
+        
+        if (!comment.getUser().equals(user)) {
+            throw new UnauthorizedAccessException("You are not authorized to delete this comment.");
+        }
```

Comment:
```suggestion
        if (!comment.getUser().getId().equals(user.getId())) {
            throw new UnauthorizedAccessException("You are not authorized to delete this comment.");
        }
```

o `comment.getUser()` tem um estrutura diferente do user (User) e por isso a comparação nunca iria ser true, iria ciar sempre no throw error da linha 78.